### ISSUE TITLE
3D Resnet for video

### ISSUE THREAD

**Issue by muna-cs** [KaiyangZhou/deep-person-reid#256]
I am trying to implement 3D Resnet for on prid2011 multishot dataset, I used:
[ this code from pytorch](https://pytorch.org/docs/stable/_modules/torchvision/models/video/resnet.html)

but I get this error:

> Expected 5-dimensional input for 5-dimensional weight 64 3 7 7, but got 4-dimensional input of size [300, 3, 128, 64] instead

How can I solve this error?

**Comment by KaiyangZhou** [comment]
3D CNN is supposed to take as input a 5-d tensor, i.e. (batch, channel, seq, height, width), so what you need to do is to change the input shape somewhere near [here](https://github.com/KaiyangZhou/deep-person-reid/blob/master/torchreid/engine/video/softmax.py#L69).

**Comment by gceboh** [comment]
for reference:
https://github.com/jiyanggao/Video-Person-ReID/blob/master/models/resnet3d.py

**Comment by KaiyangZhou** [comment]
@gceboh thanks

@muna-cs this is what I meant https://github.com/jiyanggao/Video-Person-ReID/blob/master/models/resnet3d.py#L190

**Comment by muna-cs** [comment]
I changed this line
https://github.com/KaiyangZhou/deep-person-reid/blob/099b0ae7fcead522e56228860221a4f8b06cdaad/torchreid/engine/video/softmax.py#L79
to            

>  imgs = imgs.view(b , c, s, h, w)

and this line
https://github.com/KaiyangZhou/deep-person-reid/blob/099b0ae7fcead522e56228860221a4f8b06cdaad/torchreid/engine/video/softmax.py#L92
to

>   input = input.view(b ,c, s, h, w)

but I get this error:
`Expected tensor [32, 400] and index [480, 1] to have the same size in dimension 1`
on 
https://github.com/KaiyangZhou/deep-person-reid/blob/099b0ae7fcead522e56228860221a4f8b06cdaad/torchreid/losses/cross_entropy_loss.py#L47
